### DISCUSSION TITLE
Celery redis reduce freqeuncy of BRPOP messages

### DISCUSSION THREAD

**Discussion by Jackevansevo** [celery/celery#8644] [General]
I'm running my worker with the following arguments `--without-gossip --without-mingle --without-heartbeat` with Redis as my celery broker

I can see my redis instance is frequently receiving messages (once a second) that look as follows:

```
1700056088.098427 [0 [::1]:45462] "BRPOP" "celery" "celery\x06\x163" "celery\x06\x166" "celery\x06\x169" "1"
1700056089.104080 [0 [::1]:45462] "BRPOP" "celery" "celery\x06\x163" "celery\x06\x166" "celery\x06\x169" "1"
1700056090.109984 [0 [::1]:45462] "BRPOP" "celery" "celery\x06\x163" "celery\x06\x166" "celery\x06\x169" "1"
1700056091.115298 [0 [::1]:45462] "BRPOP" "celery" "celery\x06\x163" "celery\x06\x166" "celery\x06\x169" "1"
1700056092.122189 [0 [::1]:45462] "BRPOP" "celery" "celery\x06\x163" "celery\x06\x166" "celery\x06\x169" "1"
1700056093.126348 [0 [::1]:45462] "BRPOP" "celery" "celery\x06\x163" "celery\x06\x166" "celery\x06\x169" "1"
```

This amount of traffic means I can't use certain free-tiers in cloud providers. Realistically I only need to run a single periodic task every hour (like a cronjob). Is there a way to reduce the frequency / interval of these messages?

**Comment by agronick** [comment]
The documentation for brpop says `it blocks the connection when there are no elements to pop from any of the given lists`. It doesn't seem like it is blocking at all.

**Comment by jcondominas** [comment]
Hello! I've just started using one instance from upstash, and also noticed the same behaviour, every second I see one BRPOP command. This is making me impossible to use upstash, as I'm building a simple project, and I do not want to be paying 20-30 dollars from an idle group of celery workers.

Any progress, or configuration setting I could tweak to solve this situation? I have a very basic setup, perhaps I configured something wrong, but I don't know what it could be... I just followed some documents from Celery website.

Thanks!

**Comment by oliverhaas** [comment]
Kombu 5.6.x now honors the `polling_interval` setting, see https://docs.celeryq.dev/projects/kombu/en/stable/reference/kombu.transport.redis.html.

**Reply by Alek2077** [reply]
@oliverhaas Thanks for the info. Much appreciated.
